### PR TITLE
Create a ChoiceWithOtherView which takes a choice view and an otherKe…

### PR DESCRIPTION
**Pull request by mcarcaso**
…y and shows another input if otherKey is selected.

image

image

### CODE REVIEW

File: src/foam/u2/view/ChoiceWithOtherView.js [hunk 1]
```diff
@@ -0,0 +1,98 @@
+/**
+ * @license
+ * Copyright 2019 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
```
Comment (adamvy) [inline src/foam/u2/view/ChoiceWithOtherView.js:7]: In the future I think we can just inline this into the ChoiceView as an optional feature.

**Comment by TW80000** [comment]
This only really works for string type properties, right? I don't think that's an issue, just wanted to point it out since the regular choice view can be used for more than just strings.

**Comment by mcarcaso** [comment]
> This only really works for string type properties, right? I don't think that's an issue, just wanted to point it out since the regular choice view can be used for more than just strings.

I don't think so actually. You should be able to set `otherKey` to any kind of object and same goes for the `otherDefault` but if I'm being completely honest I only really tested for Strings :p

**Comment by TW80000** [comment]
> > This only really works for string type properties, right? I don't think that's an issue, just wanted to point it out since the regular choice view can be used for more than just strings.
> 
> I don't think so actually. You should be able to set `otherKey` to any kind of object and same goes for the `otherDefault` but if I'm being completely honest I only really tested for Strings :p

Yeah, never mind, I should have given it more thought. I see now that you can set the `otherView` as well.

**Pull request merged**